[irrlicht] Enable build on arm and explicitly disable on android

traversaro opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) enties are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

traversaro committed:
[irrlicht] Enable build on arm

traversaro commented:
There are two additional commits between the actual arm enabling and the version update that were done to avoid that `vcpkg.exe x-add-version irrlicht` complained, I am not sure if they are supposed to be like that or to be squashed in a single commit.

Cheney-W commented:
> I am not sure if they are supposed to be like that or to be squashed in a single commit.

There are no requirements here, both are acceptable.
openimageio:x64-linux failure is a CI baseline issue, we are fixing it.

traversaro commented:
> openimageio:x64-linux failure is a CI baseline issue, we are fixing it.

Thanks a lot!

Cheney-W commented:
@traversaro Could you please solve the conflicts? Thank you!

traversaro commented:
> @traversaro Could you please solve the conflicts? Thank you!

Done! I did it some days ago but then the CI took a long time to get green and then I forgot about this, sorry!

Cheney-W commented on scripts/ci.baseline.txt:
```diff
@@ -589,7 +589,8 @@ infoware:x64-android=fail
 intelrdfpmathlib:arm64-android=fail
 intelrdfpmathlib:x64-android=fail
-irrlicht:arm64-windows=fail
+irrlicht:arm-neon-android=fail
+irrlicht:arm64-android=fail
```
```suggestion
```
Unnecessary.

Cheney-W commented on scripts/ci.baseline.txt:
```diff
@@ -589,7 +589,8 @@ infoware:x64-android=fail
+irrlicht:arm-neon-android=fail
+irrlicht:arm64-android=fail
 irrlicht:arm-uwp=fail
 irrlicht:x64-android=fail
```
```suggestion
```
Unnecessary.

Cheney-W commented:
I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

traversaro committed:
[irrlicht] Update versions

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -496,6 +496,8 @@ intelrdfpmathlib:arm-neon-android=fail
 intelrdfpmathlib:arm64-android=fail
 intelrdfpmathlib:x64-android=fail
 irrlicht:x64-android=fail
+irrlicht:arm-uwp=fail
+irrlicht:x64-uwp=fail
```
```suggestion
```
because both android and uwp are marked as unsupported with this PR and so won't be tested in CI.

traversaro committed:
Update scripts/ci.baseline.txt

Co-authored-by: Kai Pastor <dg0yt@darc.de>

traversaro commented:
Thanks @dg0yt, I updated the PR with your suggestion.

Cheney-W commented:
Waiting for https://github.com/microsoft/vcpkg/pull/34772 to be merged.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	scripts/ci.baseline.txt

BillyONeal committed:
Fix feature baseline.

I'm using a feature baseline skip for the tools feature on Linux because I think there could be systems that don't trigger this error:

```
In file included from /mnt/vcpkg-ci/b/irrlicht/src/irrlicht-1-633b25421b.clean/tools/IrrFontTool/newFontTool/main.cpp:20:
/mnt/vcpkg-ci/b/irrlicht/src/irrlicht-1-633b25421b.clean/tools/IrrFontTool/newFontTool/CFontTool.h:19:18: fatal error: X11/Xft/Xft.h: No such file or directory
   19 |         #include <X11/Xft/Xft.h>
      |                  ^~~~~~~~~~~~~~~
compilation terminated.
```
